api/src/main/java/org/openmrs/module/bedmanagement/aop/EncounterWithBedPatientAssignmentSaveHandler.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.bedmanagement.aop;
+
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Encounter;
+import org.openmrs.User;
+import org.openmrs.annotation.Handler;
+import org.openmrs.api.handler.SaveHandler;
+import org.openmrs.module.bedmanagement.entity.BedPatientAssignment;
+import org.openmrs.module.bedmanagement.service.BedManagementService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+@Handler(supports = Encounter.class)
+public class EncounterWithBedPatientAssignmentSaveHandler implements SaveHandler<Encounter> {
+	
+	private final Log log = LogFactory.getLog(getClass());
+	
+	private final BedManagementService bedManagementService;
+	
+	@Autowired
+	public EncounterWithBedPatientAssignmentSaveHandler(
+	    @Qualifier("bedManagementService") BedManagementService bedManagementService) {
+		this.bedManagementService = bedManagementService;
+	}
+	
+	@Override
+	public void handle(Encounter encounter, User user, Date date, String s) {
+		if (encounter.getEncounterId() != null) {
+			
+			if (encounter.getVoided()) {
```

Comment:
Here, we void any bed patient assignment associated with the encounter when the encounter is voided. However, the `BedManagementDao` contains several queries that do not account for the voided field for bed patient assignment. I filed [BED-17](https://openmrs.atlassian.net/browse/BED-17) to track this.

Reply:
Unhelpful, obvious answer but throwing an exception is safer while just voiding is easier (on the end user).  I guess the argument could be made to go for what is easier on the end user, since, assumedly, a bed assignment is not a critical piece of medical data about a patient?

Reply:
There is precedent for voiding of one resource causing voiding of other related resources, right? In [EncounterServiceImpl](https://github.com/openmrs/openmrs-core/blob/5995f82acc0c6b9b86d398590b78df34388f213d/api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java#L409), voiding an encounter causes corresponding ops to be voided as well.